src/features/events_list/atoms/autoSelectEvent.ts

```diff
@@ -22,11 +20,25 @@ export const autoSelectEvent = createAtom(
         eventListResource.data.length
       ) {
         const currentEvent = getUnlistedState(currentEventAtom);
-        const currentEventNotInNewList =
-          eventListResource.data.findIndex(
-            (e) => e.eventId === currentEvent?.id,
-          ) === -1;
-        if (currentEventNotInNewList) {
+        const currentEventNotInTheList = !eventListResource.data.some(
+          (e) => e.eventId === currentEvent?.id,
+        );
+
+        // do nothing
+        if (!currentEventNotInTheList) return;
```

Comment:
it's pretty clear why early return is needed :)
Btw, i remember we discussed on FE meetup about return undefined from atom. Is it ok?

Reply:
good point, i thought it was stateless atom
i will `return state` then